nireports/tests/test_reportlets.py

```diff
@@ -258,6 +260,18 @@ def test_plot_melodic_components(tmp_path, outdir):
         out_file=no_classified,
     )
 
+    if not have_compression:
+        pytest.skip(compression_missing_msg)
```

Comment:
Since we've gotten to here, we can just return and pass rather than marking the test skipped

```suggestion
    if not have_compression:
        return
```

Reply:
I used `pytest.skip` so that it becomes clear in the log trace that this part of the test was skipped, much like what the `@pytest.mark.skipif` decorator would do. Have not checked locally or investigated the logs.

Reply:
My issue with using skip here is nullifies all the work done prior - the test will be marked as skipped despite executing `plot_melodic_components` successfully multiple times, whereas using `return` will exit the function and allow pytest to mark it as a success.

The ideal scenario is to split out the compression check into a standalone test, which then can be skipped like you're doing if no compression is available.

Reply:
Resolving as it has been fixed.